src/main/java/org/opensearch/security/privileges/PrivilegesEvaluator.java

```diff
@@ -209,9 +209,12 @@ private void setUserInfoInThreadContext(User user, Set<String> mappedRoles) {
             joiner.add(String.join(",", user.getRoles()));
             joiner.add(String.join(",", Sets.union(user.getSecurityRoles(), mappedRoles)));
             String requestedTenant = user.getRequestedTenant();
-            if (!Strings.isNullOrEmpty(requestedTenant)) {
-                joiner.add(requestedTenant);
+            if (requestedTenant == null) {
+                requestedTenant = "";
```

Comment:
common-utils has logic to ignore it if its empty: https://github.com/opensearch-project/common-utils/blob/main/src/main/java/org/opensearch/commons/authuser/User.java#L182-L184

In this Draft PR I am serializing the new user fields `isInternal` and `authDomain` into the threadcontext after requested tenant so that field needs to be present after serialization where its optional now.

I plan to add tests around this to ensure that this change does not introduce any regressions, but in order to add fields to a user and have them serialized then this value needs to be written and not only if it is present.

Reply:
okay, sounds good. And yes, regression tests are a good idea.